roles/jd-client/src/lib/channel_manager/template_message_handler.rs

```diff
@@ -0,0 +1,559 @@
+        Err(JDCError::TxDataError.into())
+    }
+
+    #[instrument(skip_all, fields (template_id = msg.template_id), level = "trace")]
+    async fn handle_request_tx_data_success(
+        &mut self,
+        msg: RequestTransactionDataSuccess<'_>,
+    ) -> Result<(), Error> {
+        info!("Received RequestTransactionDataSuccess from Template provider");
+        debug!("RequestTransactionDataSuccess: {msg:?}");
+
+        let transactions_data = msg.transaction_list;
+        let excess_data = msg.excess_data;
+
+        let (token, template_message, request_id, prevhash) =
+            self.channel_manager_data.super_safe_lock(|data| {
+                (
+                    data.allocate_tokens.take(),
+                    data.template_store.remove(&msg.template_id),
+                    data.request_id_factory.next(),
+                    data.last_new_prev_hash.clone(),
+                )
+            });
+
+        _ = self.allocate_tokens(1).await;
+        let Some(token) = token else {
+            error!("Token not found, template id: {}", msg.template_id);
+            return Err(JDCError::TokenNotFound.into());
+        };
+
+        let Some(template_message) = template_message else {
+            error!("Template not found, template id: {}", msg.template_id);
+            return Err(JDCError::TemplateNotFound(msg.template_id).into());
+        };
+
+        let mining_token = token.mining_job_token.clone();
+        let mut deserialized_outputs: Vec<TxOut> =
+            deserialize_coinbase_output(&token.coinbase_outputs.to_vec());
+        deserialized_outputs[0].value =
+            Amount::from_sat(template_message.coinbase_tx_value_remaining);
+        let reserialized_outputs = consensus::serialize(&deserialized_outputs);
+
+        let tx_list: Vec<Transaction> = transactions_data
+            .to_vec()
+            .iter()
+            .map(|raw_tx| consensus::deserialize(raw_tx).expect("invalid tx"))
+            .collect();
+
+        let txids_as_u256: Vec<U256<'static>> = tx_list
+            .iter()
+            .map(|tx| {
+                let txid = tx.compute_txid();
+                let byte_array: [u8; 32] = *txid.as_byte_array();
+                U256::Owned(byte_array.to_vec())
+            })
+            .collect();
+
+        let tx_ids = Seq064K::new(txids_as_u256).map_err(JDCError::BinarySv2)?;
+        let is_activated_future_template = template_message.future_template
+            && prevhash
+                .map(|prev_hash| prev_hash.template_id != template_message.template_id)
+                .unwrap_or(true);
+
+        let declare_job = self.channel_manager_data.super_safe_lock(|data| {
+            let job_factory = data.job_factory.as_mut()?;
+            let output = deserialize_coinbase_output(&data.coinbase_outputs);
+
+            if let Ok((coinbase_prefix, coinbase_suffix)) =
+                job_factory.new_coinbase_tx_prefix_and_suffix(template_message.clone(), output)
```

Comment:
```suggestion
            let mut outputs = deserialize_coinbase_output(&data.coinbase_outputs);
            outputs[0].value = Amount::from_sat(template_message.coinbase_tx_value_remaining);

            if let Ok((coinbase_prefix, coinbase_suffix)) =
                job_factory.new_coinbase_tx_prefix_and_suffix(template_message.clone(), outputs)
```

`new_coinbase_tx_prefix_and_suffix` is agnostic about allocation template revenue, that's up to the caller to do

this is what was causing the rejection of block propagated by JDS with:
```
2025-08-30T15:52:51Z [error] ProcessNewBlock: AcceptBlock FAILED (high-hash, proof of work failed)
```

Reply:
https://github.com/stratum-mining/stratum/pull/1849 should mitigate this API footgun